server/azureAppInsights.ts

```diff
@@ -0,0 +1,49 @@
+import { setup, Contracts, defaultClient, DistributedTracingModes } from 'applicationinsights'
+import type { Request } from 'express'
+import applicationVersion from './applicationVersion'
+import config from './config'
+import logger from '../log'
+
+function ignoreStaticAssetsProcessor(
+  envelope: Contracts.EnvelopeTelemetry,
+  _contextObjects: { [name: string]: unknown } | undefined
+): boolean {
+  if (envelope.data.baseType === Contracts.TelemetryTypeString.Request) {
+    const requestData = envelope.data.baseData
+    if (requestData instanceof Contracts.RequestData && requestData.name.startsWith('GET /assets/')) {
+      return false
+    }
+  }
+  return true
+}
+
+function addUsernameProcessor(
+  envelope: Contracts.EnvelopeTelemetry,
+  contextObjects: { [name: string]: unknown } | undefined
+): boolean {
+  if (envelope.data.baseType === Contracts.TelemetryTypeString.Request) {
+    const username = (contextObjects?.['http.ServerRequest'] as Request)?.user?.username
+    if (username) {
+      // eslint-disable-next-line no-param-reassign
+      envelope.tags['ai.user.authUserId'] = username
```

Comment:
🔐 `username` contains our PPs' real names; could we use the user ID instead, please?

Reply:
done !